wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,256 @@
+# Scroll View
+
+## Summary
+
+This spec proposes to add an experimental native ScrollView component to Electron.
+
+## Platforms
+* macOS 
+* Windows
+* Linux
+
+## Impetus
+
+It is currently not possible to add scrolling functionality around BrowserViews:
+
+* Capture mouse / touch events
+* Reposition BrowserViews accordingly
+
+More details of the use-case and challenges related to workarounds: [#32751](https://github.com/electron/electron/issues/32751)
+
+![](https://user-images.githubusercontent.com/2891206/150981006-84fe796e-60bd-43c0-952d-831d1bfd9998.gif)
+
+## Usage Example
+
+```js
+const { BrowserView, BrowserWindow, ContainerView, ScrollView, WrapperBrowserView } = require("electron");
+
+const window = new BrowserWindow({ width: 1000, height: 500 })
+
+const scroll = new ScrollView()
+scroll.setBounds({ width: 1000, height: 500 })
+
+const scrollContent = new ContainerView()
+scrollContent.setBounds({ width: 2000, height: 500 })
+
+const browserView1 = new BrowserView()
+const wrapperBrowserView1 = new WrapperBrowserView({ 'browserView': browserView1 });
+wrapperBrowserView1.setBounds({ width: 1000, height: 500 })
+scrollContent.addChild(wrapperBrowserView1)
+
+const browserView2 = new BrowserView()
+const wrapperBrowserView2 = new WrapperBrowserView({ 'browserView': browserView2 });
+wrapperBrowserView2.setBounds({ width: 1000, height: 500, x: 1000 })
+scrollContent.addChild(wrapperBrowserView2)
+
+scroll.setContentView(scrollContent);
+win.addChild(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+## BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
+
+Process: **Main**
+
+`BaseView` is an EventEmitter
+
+### `new BaseView()`
+
+Creates the new base view.
+
+### Instance Methods
+
+#### `view.setBounds(bounds)`
+
+Resizes and moves the view to the supplied bounds relative to its parent.
+
+#### `view.getBounds()`
+
+Returns the position and size of the view.
+
+#### `view.setBackgroundColor(color)`
+
+Change the background color of the view.
+
+#### `view.getParent()`
+
+Returns `BaseView || null` - The parent view, otherwise returns `null`.
+
+#### `view.getWindow()`
+
+Returns `BrowserWindow || null` - The window that the view belongs to, otherwise returns `null`.
+
+### Events
+
+#### Event: 'size-changed'
+
+Emitted when the view's size has been changed.
+
+ 
+## BrowserWindow 
+
+### New instance methods
+
+#### `win.addChild(view)`
+
+#### `win.removeChild(view)`
```

Comment:
> What happens if passed in a view that is not a child -- is there any mechanism for the caller to know that this failed?

For now exactly the same behaviour as in case of BaseWindow.removeBrowserView - just does nothing

> Can this be used to remove any descendant view, or just direct children?

Just direct children